File: ocaml/tornado.ml
let shift str = String.sub str 1 (String.length str - 1) ^ String.make 1 (String.get str 0)

let rec loop text = function
  | 0 -> ()
  | n -> print_endline text;
         loop (shift text) (n - 1)

let text =
  print_string "Enter some text: ";
  read_line ()

let num =
  print_string "Enter line amount: ";
  try read_int () with _ -> ~-1

let () = loop text (num + 1)

File: rust/tornado.rs
use std::io;


fn shift_and_print(txt: String) -> String {
	println!("{}", txt);
	let mut temp = String::from(txt);
	let char = temp.chars().nth(0).unwrap();
	temp.push(char);
	temp.remove(0);
	temp
}

fn main(){
	let mut input = String::new();
	let mut num = String::new();
	println!("Enter some text: ");
	match io::stdin().read_line(&mut input){
		Ok(_) => {
			println!("Enter line count (-1 for infinity)");
			match io::stdin().read_line(&mut num){
				Ok(_) => {
					let mut text = String::from(input.trim());
					let line_count_str = num.trim();
					match line_count_str.parse::<i32>(){
						Ok(line_count) => {
							if text.chars().last().unwrap() != ' ' {
								text.push(' ');
							}

							if line_count == -1 {
								loop {
									text = shift_and_print(text);
								}
							}

							for _ in 0..line_count {
								text = shift_and_print(text);
							}
						}
						Err(error) => {
							println!("error {}", error);
						}
					}
				}
				Err(error) => {
					println!("error {}", error);
					main();
				}
			}
		}
		Err(error) => {
			println!("error: {}", error);
			main();
		}
	}
}